src/Haskell/Ide/Engine/Plugin/Liquid.hs

```diff
@@ -120,7 +120,9 @@ diagnosticProvider DiagnosticOnSave uri cb = pluginGetFile "Liquid.diagnosticPro
     LiquidData mtid <- get
     mapM_ (liftIO . cancel) mtid
 
-    tid <- liftIO $ async $ generateDiagnosics cb uri file
+    let progTitle = "Running Liquid Haskell on " <> T.pack (takeFileName file)
+    tid <- withIndefiniteProgress progTitle $
+      liftIO $ async $ generateDiagnosics cb uri file
```

Comment:
This isn't right. The progress report needs to go inside the call to `async` and if you do that, you'll discover that the types for `withIndefiniteProgress` are not quite right still..

Reply:
You’re right, this completely slipped by me! I was testing it by adding a `threadDelay 10000000`…